src/main/java/Duke.java

```diff
@@ -1,10 +1,318 @@
+import exceptions.InvalidDeadlineException;
+import exceptions.InvalidEventException;
+import exceptions.InvalidValueException;
+import exceptions.InvalidTodoException;
+import exceptions.UnrecognisedCommandException;
+import tasks.Deadline;
+import tasks.Task;
+import tasks.Event;
+import tasks.Todo;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+
 public class Duke {
+
+    // HELPER FUNCTIONS
+    public static void greetUser() {
+        String line = "------------------------\n";
+        System.out.println(line + "Hello! I'm Juke!\nWhat can I do for you?\n" + line);
+    }
+
+    public static File getMakeFile() {
+        try {
+            File folder = new File("data/");
+            File file = new File("data/stored.txt");
+            
+            if (!folder.exists() && !folder.isDirectory()) {
+                // need to create folder as determined by the path string
+                folder.mkdirs();
+                file.createNewFile();
+                return file;
+            }
+            else if (!file.exists() && folder.isDirectory()) {
+                // need to create file as determined by the path string
+                file.createNewFile();
+                return file;
+            }
+            return file;
+
+        } catch (Exception e) {
+            System.out.println("File i/o error.");
+            return null;
+        }
+    }
+
+    // ArrayLists are passed by reference
+    public static void insertPreviousItems(File f, ArrayList<Task> tasks) {
+        try {
+            Scanner fileScanner = new Scanner(f);
+        
+            while (fileScanner.hasNext()) {
+                String lineInput = fileScanner.nextLine();
+                String[] words = lineInput.split("\\|", 4);
+
+                switch (words[0]) {
+                case "T":
+                    Todo newTodo = new Todo(words[2]);
+                    if (words[1].equals("Y")) {
+                        newTodo.markDone();
+                    }
+                    tasks.add(newTodo);
+                    break;
+
```

Comment:
Consider removing the whitespace between `case` statements: https://se-education.org/guides/conventions/java/basic.html#layout